File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    citations.sort() # 0 1 3 5 6 7 8
    l = len(citations)
    for i in range(l-1, -1, -1):
        if citations[i] >= l-i:
            answer = l-i
    
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.begin(), citations.end());
    int n = citations.size();
    for (int i=n;i>=0;i--){
        if (citations[i] >= n-i){
            answer = n-i;
        }
    }
    return answer;
}